srcs/class/route.cpp

```diff
@@ -1,53 +1,173 @@
 #include "route.hpp"
-#include "../webserv.hpp"
 
-Route::Route( std::string location, std::string root ) {
+// default route
+Route::Route(std::string location, std::string root)
+{
 
 	this->root = root;
 	this->location = location;
 	index.push_back("index.html");
 	methods.push_back("GET");
+	this->auto_index = true;
+	this->cgi_enable = false;
 };
 
-Route::Route( void ) {
-
+Route::Route(Route rhs, int notcopy)
+{
+	(void)notcopy;
+	this->root = finish_by_only_one(rhs.root, '/');
+	this->location = finish_by_only_one(rhs.location, '/');
 	this->auto_index = true;
 	this->cgi_enable = false;
 	index.push_back("index.html");
-	methods.push_back("GET");
-}
+	methods.push_back("GET");	
+};
 
-Route::Route( const Route &rhs ) {
+Route::Route(std::string location, std::string root, int notdefault)
+{
+	(void)notdefault;
+	this->root = root;
+	this->location = location;
+};
 
-	this->root = finish_by_only_one( rhs.root, '/' );
-	this->location = finish_by_only_one( rhs.location, '/' );
-	this->auto_index = true;
-	this->cgi_enable = false;
-	this->cgi_path = rhs.cgi_path;
+Route::Route(const Route &rhs)
+{
+	this->root = rhs.root;
+	this->location = rhs.location;
+	this->auto_index = rhs.auto_index;
+	this->cgi_enable = rhs.cgi_enable;
 	this->cgi_extension = rhs.cgi_extension;
+	this->cgi_path = rhs.cgi_path;
+
+	this->index = rhs.index;
+	this->methods = rhs.methods;
 	this->error_pages = rhs.error_pages;
 	this->redirections = rhs.redirections;
-	index.push_back("index.html");
-	methods.push_back("GET");
 };
 
-Route::~Route(void) {};
 
-void	Route::enable_cgi( std::string path ) {
+Route& Route::operator=(const Route& rhs)
+{
+	this->root = rhs.root;
+	this->location = rhs.location;
+	this->auto_index = rhs.auto_index;
+	this->cgi_enable = rhs.cgi_enable;
+	this->cgi_extension = rhs.cgi_extension;
+	this->cgi_path = rhs.cgi_path;
+
+	this->index = rhs.index;
+	this->methods = rhs.methods;
+	this->error_pages = rhs.error_pages;
+	this->redirections = rhs.redirections;
+	return (*this);
+}
+
+
+Route::~Route(void){};
```

Comment:
Pour uniformiser un peu le code !

```suggestion
Route::~Route(void)
{}
```